pandas/core/categorical.py

```diff
@@ -1735,31 +1737,34 @@ def convert_categorical(x):
             return x.get_values()
         return x.ravel()
 
-    typs = get_dtype_kinds(to_concat)
-    if not len(typs-set(['object','category'])):
-
-        # we only can deal with object & category types
-        pass
-
-    else:
-
+    if get_dtype_kinds(to_concat) - set(['object', 'category']):
```

Comment:
why not? we have

1. everything is a categorical because the filtering above when building up `categoricals`
2. all the categories are equivalent via `categories.is_dtype_equal(first_set_of_categories)`

what else is there to check?